File: TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/services/RespuestaAutenticacion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.services;

import ec.edu.monster.modelo.Usuario;

/**
 *
 * @author josue
 */
public class RespuestaAutenticacion {
    private boolean exitoso;
    private String mensaje;
    private String username;
    
    public RespuestaAutenticacion() {
    }
    
    public RespuestaAutenticacion(boolean exitoso, String mensaje) {
        this.exitoso = exitoso;
        this.mensaje = mensaje;
    }
    
    public RespuestaAutenticacion(boolean exitoso, String mensaje, Usuario usuario) {
        this.exitoso = exitoso;
        this.mensaje = mensaje;
        if (usuario != null) {
            this.username = usuario.getUsername();
        }
    }
    
    // Getters y Setters
    public boolean isExitoso() {
        return exitoso;
    }
    
    public void setExitoso(boolean exitoso) {
        this.exitoso = exitoso;
    }
    
    public String getMensaje() {
        return mensaje;
    }
    
    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }
    
    public String getUsername() {
        return username;
    }
    
    public void setUsername(String username) {
        this.username = username;
    }
}

File: TI1.4 RESTFUL_DOTNET_SINBDD_GR06/ConUni_RestFul_Dotnet_GR06/ConUni_RestFul_Dotnet_GR06/Modelos/RespuestaAutenticacion.cs
namespace ConUni_RestFul_Dotnet_GR06.Modelos
{
    /// <summary>
    /// DTO para respuestas de autenticación
    /// </summary>
    public class RespuestaAutenticacion
    {
        public bool Exitoso { get; set; }
        public string Mensaje { get; set; }
        public string Username { get; set; }

        public RespuestaAutenticacion()
        {
        }

        public RespuestaAutenticacion(bool exitoso, string mensaje)
        {
            Exitoso = exitoso;
            Mensaje = mensaje;
        }

        public RespuestaAutenticacion(bool exitoso, string mensaje, Usuario usuario)
        {
            Exitoso = exitoso;
            Mensaje = mensaje;
            if (usuario != null)
            {
                Username = usuario.Username;
            }
        }
    }
}
